### PR TITLE
fix(AutoModerationActionExecution): transform `action`

### LINKED ISSUE #9110: Typing of AutoModerationActionMetadata incorrect
### Which package is this bug report for?

discord.js

### Issue description

Typing says there should be `channelId` or `durationSeconds`
There actually is `channel_id` or `duration_seconds`
i.e. parameters not transformed or typing incorrect

1. Create Automoderation Rule with timeout or alert channel
2. listen to `autoModerationActionExecution` event
3. log `<Discord.AutoModerationActionExecution>.action.metadata` to console
4. trigger the auto moderation rule

### Code sample

```typescript
const Discord = require("discord.js");

const client = new Discord.Client({
  shards: "auto",
  intents: [Discord.IntentsBitField.Flags.AutoModerationExecution],
});

client.login(
  "TOKEN"
);

client.on("autoModerationActionExecution", (execution) => {
  console.log(execution.action.metadata);
});
```

### Package version

14.7.1

### Node.js version

17.8.0

### Operating system

Ubuntu 20.04

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

AutoModerationExecution

### I have tested this issue on a development release

405f940

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
The typings did not match the code here as we were not properly parsing the returned data. This resolves #9110.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by jaw0r3k** [comment]
shouldn't it be a static function on automoderationrule?
or easier just just make a `AutoModerationAction` class

File: packages/discord.js/src/util/Util.js [hunk 1]
```diff
@@ -371,6 +371,22 @@ function parseWebhookURL(url) {
   };
 }
 
+/**
+ * Transforms an API auto moderation action object to a camel-cased variant.
+ * @param {APIAutoModerationAction} autoModerationAction The action to transform
+ * @returns {AutoModerationAction}
+ * @ignore
+ */
+function _transformAPIAutoModerationAction(autoModerationAction) {
```
Comment (vladfrangu) [inline packages/discord.js/src/util/Util.js:380]: This would probably fit better in Transformers.ts

**Review by vladfrangu — APPROVED** [review]

File: packages/discord.js/src/util/Util.js (hunk 1 above)
Comment (Jiralite) [reply packages/discord.js/src/util/Util.js]: Agreed! I've moved it with e08863afaaf2a12a391193a0b8852629aa18ebdd.

**Comment by Jiralite** [comment]
> or just easier make an `AutoModerationAction` class

I personally think this is overkill.

**Comment by jaw0r3k** [comment]
> > or just easier make an `AutoModerationAction` class
> 
> I personally think this is overkill.

also keep in mind that #8959 is coming, so it will be a bit bigger

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**